### ISSUE TITLE
support mongodb for scala3

### ISSUE THREAD

**Issue by pjfanning** [apache/incubator-pekko-connectors#150]
* blocked by https://jira.mongodb.org/browse/JAVA-4181
* one potential workaround is to change the compile dependencies to use mongodb-driver-reactivestreams instead of mongo-scala-driver
  * the runtime code does not use the Scala specific mongodb classes
  * the Scala specific mongodb are used in tests though - the scala tests could be removed or only run with Scala2
  * Scala 3 users who want to use the mongodb connector would need to use Java classes to bind because mongo-scala-driver has the code to support case classes, etc.

**Comment by bdoyle0182** [comment]
maybe not the right place to chime in, but would porting `reactivemongo-akkastream` to have a pekko version apply here? I'm trying to coordinate on getting a pekko version of `akka-persistence-mongo` which was never an official akka connector. The one major blocker I can see is it depends on `reactivemongo-akkastream` otherwise the only other thing it uses is akka core

**Comment by pjfanning** [comment]
> maybe not the right place to chime in, but would porting `reactivemongo-akkastream` to have a pekko version apply here? I'm trying to coordinate on getting a pekko version of `akka-persistence-mongo` which was never an official akka connector. The one major blocker I can see is it depends on `reactivemongo-akkastream` otherwise the only other thing it uses is akka core which is now already covered by pekko core 1.0.0

If you or anyone else wants to fork reactivemongo to support pekko or submit a patch to reactivemongo that would be very helpful.

My personal preference would be to do this outside of the Pekko project - which has plenty of code to manage as it is.

**Comment by pjfanning** [comment]
I raised https://github.com/ReactiveMongo/ReactiveMongo-Streaming/issues/320

**Comment by bdoyle0182** [comment]
I started working on adding a pekkostream module to the reactivemongo streaming repo in a fork. I think I've done everything I can, pretty straight forward, but since the core project this repo is dependent on is built using akka it would require cutting over the core reactivemongo repo to pekko and buy in from the reactivemongo team.

**Comment by mdedetrich** [comment]
> I started working on adding a pekkostream module to the reactivemongo streaming repo in a fork. I think I've done everything I can, pretty straight forward, but since the core project this repo is dependent on is built using akka it would require cutting over the core reactivemongo repo to pekko and buy in from the reactivemongo team.

So the current pekko-connectors mongo driver has been udpated to work within Pekko but I think going forward it works out it makes sense to support a pekko version of reactive-mongo as its own project in the same vain as how we have pekko-connectors-kafka.

**Comment by pjfanning** [comment]
> > I started working on adding a pekkostream module to the reactivemongo streaming repo in a fork. I think I've done everything I can, pretty straight forward, but since the core project this repo is dependent on is built using akka it would require cutting over the core reactivemongo repo to pekko and buy in from the reactivemongo team.
> 
> So the current pekko-connectors mongo driver has been udpated to work within Pekko but I think going forward it works out it makes sense to support a pekko version of reactive-mongo as its own project in the same vain as how we have pekko-connectors-kafka.

This is covered by https://github.com/ReactiveMongo/ReactiveMongo/pull/1245

**Comment by mdedetrich** [comment]
> This is covered by [ReactiveMongo/ReactiveMongo#1245](https://github.com/ReactiveMongo/ReactiveMongo/pull/1245)

I see, I had a quick look at the project and it just provides the basic driver however what the current pekko-connectors mongodb provides is also Pekko Stream integration (i.e. `Source`/`Sink`/`Flow`) so once this is released then we should either update the current pekko-connectors mongodb module to use reactive mongo (as a breaking change) or my previous suggestion which would be to create a pekko-connectors-mongodb which uses ReactiveMongo underneath and deprecate the current mongodb module

**Comment by pjfanning** [comment]
Yes, a new connector and deprecating the existing one may be a tidier approach

**Comment by mdedetrich** [comment]
> Yes, a new connector and deprecating the existing one may be a tidier approach

Agreed, although I think the determining factor is whether the Stream API would be any different, i.e. it might be that the structure and types of the `Flow`/`Sink`/`Source` would be the same even with the reactive mongo driver in which case it might be fine to just do a major update in the current pekko-connectors mongo module
